### PR TITLE
Code Refactoring

### PR DESCRIPTION
**Pull request by deye9**
Code refactoring

### CODE REVIEW

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -1,74 +1,74 @@
...
-        logger.info("AuthenticationController.registerUser() called with request: " + request);
+        log.info("AuthenticationController.registerUser() called with request: " + request);
 
         try {
             User registedUser = service.register(request);
 
-            return ResponseHandler.generateResponse(null, registedUser, HttpStatus.CREATED);
+            return ResponseEntity.status(HttpStatus.CREATED)
+                    .body(Collections.singletonMap(ResponseKeys.data.toString(), registedUser));
         } catch (Exception e) {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:46]: It's bad Practice to catch `Èxception` class

File: services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+@RestController
+@CrossOrigin
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/users/roles")
+public class RoleController {
+
+    private final RoleService roleService;
+
+    @PostMapping(produces = "application/json", consumes = "application/json")
+    public ResponseEntity<Map<String, ?>> createRole(@Valid @RequestBody Role role) {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java:32]: You probably shouldn't be using wildcards like this, this will increase your code's running time

File: services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+        try {
+            Optional<Role> role = roleService.findByIdAndDeletedAtIsNull(id);
+
+            return role.<ResponseEntity<Map<String, ?>>>map(
+                    value -> ResponseEntity.ok().body(Collections.singletonMap(ResponseKeys.data.toString(), value)))
+                    .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND)
+                            .body(Collections.singletonMap(ResponseKeys.data.toString(), "Role not found")));
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java:59]: You should avoid returning this HTTP status to users as much as possible

File: services/user-management/src/main/java/com/foodie/user/jwt/SecurityConfiguration.java
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/jwt/SecurityConfiguration.java:1]: You have a major concern on line 47

File: services/user-management/src/main/java/com/foodie/user/model/Token.java
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/model/Token.java:1]: Why are the fields in this class public

File: services/user-management/src/main/java/com/foodie/user/model/User.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
 @AllArgsConstructor
 @Entity
 @Table(name = "users")
-public class User extends FoodieBaseModel implements UserDetails {
+public class User extends FoodieBaseEntity implements UserDetails {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/model/User.java:31]: You're keeping the `implements userDetails`?

File: services/user-management/src/main/java/com/foodie/user/model/UserRole.java
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/model/UserRole.java:1]: Why isn't this class extending the BaseEntity class?

File: services/user-management/src/main/java/com/foodie/user/model/repositories/RoleRepository.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.foodie.user.model.repositories;
+
+// import java.util.Optional;
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/model/repositories/RoleRepository.java:3]: You'll probably want to remove this line and others

File: services/user-management/src/main/java/com/foodie/user/model/service/impl/RoleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import com.foodie.FoodieBaseRepositoryImpl;
+import com.foodie.user.model.Role;
+import com.foodie.user.model.repositories.RoleRepository;
+import com.foodie.user.model.service.RoleService;
+
+@Service
+@Transactional
+public class RoleServiceImpl extends FoodieBaseRepositoryImpl<Role, UUID> implements RoleService {
```
Comment (sfagade) [inline services/user-management/src/main/java/com/foodie/user/model/service/impl/RoleServiceImpl.java:15]: I'm not sure what the plan is here, but this smells bad

File: services/user-management/src/test/java/com/foodie/user/controllers/RoleControllerTest.java
Comment (sfagade) [inline services/user-management/src/test/java/com/foodie/user/controllers/RoleControllerTest.java:1]: You can remove all the `public` modifiers and unwanted code

**Review by sfagade — CHANGES_REQUESTED** [review]
Less review comments this time

File: models/src/main/java/com/foodie/FoodieBaseRepository.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.foodie;
+
+import java.io.Serializable;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.repository.NoRepositoryBean;
+
+@NoRepositoryBean
+public interface FoodieBaseRepository<T extends FoodieBaseEntity, ID extends Serializable>
```
[the remark this answers is not in the document]
Comment (deye9) [reply models/src/main/java/com/foodie/FoodieBaseRepository.java:9]: Followed this guide  [https://stackoverflow.com/questions/19417670/using-generics-in-spring-data-jpa-repositories]but not sure how to fix it. Guidance needed sir.

File: models/src/main/java/com/foodie/FoodieBaseRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.util.Streamable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public abstract class FoodieBaseRepositoryImpl<T extends FoodieBaseEntity, ID extends Serializable>
```
[the remark this answers is not in the document]
Comment (deye9) [reply models/src/main/java/com/foodie/FoodieBaseRepositoryImpl.java:18]: It compiled.

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java [hunk 9]
```diff
@@ -1,74 +1,74 @@
...
-    @Autowired
     private final AuthenticationService service;
 
-    private final Logger logger = LoggerFactory.getLogger(RoleController.class);
-
     @PostMapping(path = "/register", produces = "application/json", consumes = "application/json")
     public ResponseEntity<?> registerUser(@Valid @RequestBody RegisterRequest request) {
 
-        logger.info("AuthenticationController.registerUser() called with request: " + request);
+        log.info("AuthenticationController.registerUser() called with request: " + request);
```
[the remark this answers is not in the document]
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:39]: Noted.

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java (hunk 1 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:46]: Ok. What should I catch here

File: services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java (hunk 2 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java:32]: I had to do that because of the exception that am catching. Am not sure there is a way to inform it to return either one of two types like we do in typescript

File: models/src/main/java/com/foodie/enums/ResponseKeys.java [hunk 10]
```diff
@@ -0,0 +1,8 @@
+package com.foodie.enums;
+
+public enum ResponseKeys {
+    info,
```
[the remark this answers is not in the document]
Comment (deye9) [reply models/src/main/java/com/foodie/enums/ResponseKeys.java:4]: Fixed

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java [hunk 11]
```diff
@@ -1,74 +1,74 @@
...
 
         try {
             User registedUser = service.register(request);
```
[the remark this answers is not in the document]
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:42]: @RequiredArgsConstructor fixes that I believe. Working with no issues so far.

File: services/user-management/src/main/java/com/foodie/user/jwt/SecurityConfiguration.java
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/jwt/SecurityConfiguration.java:1]: private final JwtAuthenticationFilter jwtAuthFilter; is what is on line 47. Shed more light please

File: services/user-management/src/main/java/com/foodie/user/model/Token.java
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/model/Token.java:1]: Fixed.

File: services/user-management/src/main/java/com/foodie/user/model/User.java (hunk 4 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/model/User.java:31]: Working on moving it. Still not certain the best way yet.

File: services/user-management/src/main/java/com/foodie/user/model/UserRole.java
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/model/UserRole.java:1]: https://stackoverflow.com/questions/19417670/using-generics-in-spring-data-jpa-repositories

File: services/user-management/src/main/java/com/foodie/user/model/repositories/RoleRepository.java (hunk 5 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/model/repositories/RoleRepository.java:3]: Missed it previously.

File: services/user-management/src/main/java/com/foodie/user/model/service/impl/RoleServiceImpl.java (hunk 6 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/model/service/impl/RoleServiceImpl.java:15]: https://stackoverflow.com/questions/19417670/using-generics-in-spring-data-jpa-repositories

File: services/user-management/src/test/java/com/foodie/user/controllers/RoleControllerTest.java
Comment (deye9) [reply services/user-management/src/test/java/com/foodie/user/controllers/RoleControllerTest.java:1]: Done

File: models/src/main/java/com/foodie/FoodieBaseRepository.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (sfagade) [reply models/src/main/java/com/foodie/FoodieBaseRepository.java:9]: Generally, when I go to Stackoverflow I follow the post that's marked with the green marker. You seem to have follow the post which seem more recent, I think that design is overly complicated, hard to follow and it's going to be harder for newbies to understand.

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java (hunk 1 above)
Comment (sfagade) [reply services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:46]: Ideally, you'll have a custom-exception class which will extends maybe RuntimeException class and then catch that custom class. Or just straighout catch RuntimeException here

File: services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java (hunk 2 above)
Comment (sfagade) [reply services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java:32]: You can have classes which represent the possible return types from this method and have those classes extends a base class. Then you'll make that base class the wildcard here

File: services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java (hunk 3 above)
Comment (sfagade) [reply services/user-management/src/main/java/com/foodie/user/controllers/RoleController.java:59]: Yes, you should always return something that provides user friendly response to the caller and shows the other developers that you're handling every known contingencies.

**Review by sfagade — CHANGES_REQUESTED** [review]
Answered four comments

File: models/src/main/java/com/foodie/FoodieBaseRepository.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (deye9) [reply models/src/main/java/com/foodie/FoodieBaseRepository.java:9]: Will make some adjustment to it down the line

File: services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java (hunk 1 above)
Comment (deye9) [reply services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java:46]: Noted
